src/ContactPeople/ContactPeopleForm.js

```diff
@@ -1,193 +1,104 @@
+    this.removeItem = this.removeItem.bind(this);
   }
 
-  onChangeSelect = (e, elem, propertyName) => {
-    const { dispatch, change } = this.props;
-    dispatch(change(`${elem}.${propertyName}`, e));
-  }
-
-  selectedValues = (index, fields, propertyName) => {
-    const { stripes: { store } } = this.props;
+  static getDerivedStateFromProps(props, state) {
+    const { parentMutator, stripes: { store } } = props;
     const formValues = getFormValues('FormVendor')(store.getState());
-    const currValues = formValues[fields.name][index][propertyName];
-    return currValues;
-  }
+    const contactArr = formValues.contacts;
+    const queryContacts = (arr) => {
+      let newQuery = 'query=(id=null)';
+      if (arr.length >= 1) {
+        const items = arr.map(item => {
+          return `id="${item}"`;
+        });
+        const biuldQuery = items.join(' or ');
+        newQuery = `query=(${biuldQuery})`;
+      }
+      return parentMutator.queryCustom.update({ contactIDs: newQuery });
+    };
 
-  onPhoneStateUpdate(obj) {
-    if (!obj) return false;
-    return this.setState(obj);
+    if (!isEqual(contactArr, state.contactArr)) {
+      queryContacts(contactArr);
+      return { contactArr };
+    }
+    return null;
   }
 
-  // Multi Select
-  toString = (option) => option;
-  formatter = ({ option }) => {
-    const { dropdownVendorCategories } = this.props;
-    const item = _.find(dropdownVendorCategories, { id: option }) || option;
-    if (!item) return option;
-    return <div>{item.value}</div>;
-  };
+  removeItem(index) {
+    this.fields.remove(index);
+  }
 
-  filterItems = (filterText, list) => {
-    const filterRegExp = new RegExp(`^${filterText}`, 'i');
-    const renderedItems = filterText ? list.filter(item => item.search(filterRegExp) !== -1) : list;
-    return { renderedItems };
-  };
-  // End Multi Select
+  renderData(valueID) {
+    const { parentResources } = this.props;
+    const contacts = ((parentResources || {}).contacts || {}).records || [];
+    if (contacts.length === 0) return null;
+    const item = find(contacts, { id: valueID });
+    if (isEmpty(item)) return null;
+    const fullName = `${get(item, 'prefix', '')} ${get(item, 'first_name', '')} ${get(item, 'last_name', '')}`;
+    return (
+      <Fragment>
+        {fullName}
+      </Fragment>
+    );
+  }
 
-  renderCreateContact = ({ fields }) => {
+  listItem(item, index) {
+    const valueID = this.fields.get(index);
     return (
-      <Row>
-        {fields.length === 0 &&
-          <Col xs={12}>
-            <div><em>{<FormattedMessage id="ui-vendors.contactPeople.pleaseAddContactPerson" />}</em></div>
-          </Col>
-        }
-        {fields.map(this.renderSubCreateContact)}
-        <Col xs={12} style={{ paddingTop: '10px' }}>
-          <Button onClick={() => fields.push({})}>{<FormattedMessage id="ui-vendors.contactPeople.add" />}</Button>
-        </Col>
-      </Row>
+      <div>
+        <li key={index}>
```

Comment:
you don't need `key` prop here, since `div` is  a wrapper. IMO, the key should be there, if needed